torch/_subclasses/meta_utils.py

```diff
@@ -809,13 +809,16 @@ def _empty_create_subclass(
                 # We are hitting plain meta_desc tensor so actually
```

Comment:
I have an alternate proposal: just make this function call recursively back to `meta_tensor`, which as you say, handles caching. Does that work?

Reply:
agreed with this! this feels like the right thing to do and I'm curious if there's some obscure reason it wasn't

Reply:
I'm pretty sure there's no good reason, there are a bunch of places where we don't recurse back to top level when we should. It's been on my TODO list to force all the recurse-y calls to actually recurse, but I never got around to it.